components/places/src/ffi.rs

```diff
@@ -43,6 +48,155 @@ impl UniffiCustomTypeWrapper for Url {
     }
 }
 
+impl UniffiCustomTypeWrapper for Guid {
+    type Wrapped = String;
+
+    fn wrap(val: Self::Wrapped) -> uniffi::Result<Guid> {
+        Ok(Guid::new(val.as_str()))
+    }
+
+    fn unwrap(obj: Self) -> Self::Wrapped {
+        obj.into()
+    }
+}
+
+impl UniffiCustomTypeWrapper for Timestamp {
+    type Wrapped = i64;
+
+    fn wrap(val: Self::Wrapped) -> uniffi::Result<Timestamp> {
+        Ok(Timestamp::from(val as u64))
+    }
+
+    fn unwrap(obj: Self) -> Self::Wrapped {
+        obj.as_millis() as i64
+    }
+}
+
+///*
+/// A bookmark node.
+///
+/// We use a single message type for bookmarks. It covers insertion, deletion,
+/// and update, and represents all three bookmark types.
+///
+/// This simplifies the FFI by reducing the number of types that must go across
+/// it, and retuces boilderplate, but removes some static-ish guarantees we
+/// might have otherwise.
+///
+/// Note that these docs comments are internal, and don't necessarily impact the actual
+/// API we expose to Kotlin/Swift (this is particularly true around reads).
+#[derive(Clone, PartialEq)]
+pub struct BookmarkNode {
```

Comment:
I'm a little confused by the relationship between `BookmarkNode` and `PublicNode` but I suspect that will get clearer soon!

Reply:
Actually, this is a good catch. Initially I moved `BookmarkNode` here before I decided I was going to expose `PublicNode`. `BookmarkNode` isn't being used here so I'm going to remove it.